src/vm/reflectioninvocation.cpp

```diff
@@ -337,6 +337,15 @@ FCIMPL7(void, RuntimeFieldHandle::SetValue, ReflectFieldObject *pFieldUNSAFE, Ob
 
     FieldDesc* pFieldDesc = gc.refField->GetField();
 
+    // Verify we're not trying to set the value of a static initonly field
```

Comment:
When you have a choice, it is better to have code inside `HELPER_METHOD_FRAME_BEGIN_PROTECT`. It reduces chances of GC holes. (You will need to use `COMPlusThrow` once you do that.)